### PR TITLE
Improved admin interface for course_metadata models

### PR DESCRIPTION
**Pull request by clintonb**
Added listing displays and search to relevant models.

[ECOM-3765](https://openedx.atlassian.net/browse/ECOM-3765)

### CODE REVIEW

**Comment by clintonb** [comment]
@peter-fogg @bderusha

File: course_discovery/apps/course_metadata/admin.py [hunk 1]
```diff
@@ -19,15 +19,43 @@ class SeatInline(admin.TabularInline):
...
+
+
+# Register key-name mdoels
+models = (Organization, Person,)
+
+for model in models:
+    admin.site.register(model, KeyNameAdmin)
+
+# Register children of AbstractNamedModel
+models = (LevelType, Subject, Prerequisite,)
```
Comment (peter-fogg) [inline course_discovery/apps/course_metadata/admin.py]: Reusing variable names hurts readability. Can we call this `named_models`, `key_name_models`, ~~`prune_models`~~ `basic_models`?

Comment (clintonb) [inline course_discovery/apps/course_metadata/admin.py]: Variables removed. Also, I see what you did there!

**Comment by peter-fogg** [comment]
:+1:

**Pull request merged**